### ISSUE TITLE
Accessibility issue with Image module's form output

### ISSUE THREAD

**Issue by wesruv** [backdrop/backdrop-issues#1114]
The `label`'s for is "edit-field-logo-und-0", while the `input`'s `id` is edit-field-logo-und-0**-upload**
:disappointed: 

Found using WAVE accessibility toolbar on Backdropcms.org
![image](https://cloud.githubusercontent.com/assets/5607236/9429945/d85ad7f6-49af-11e5-8357-4fd47da8cdad.png)

**Comment by Dinornis** [comment]
I just checked the latest dev and it looks like this is still relevant.

**Comment by olafgrabienski** [comment]
Yes, I can reproduce the issue as well. (Had a look at the image field of the *Post* content type).

@wesruv Do you have an idea how (or where) to fix it?
